app/src/main/java/com/hedvig/app/feature/embark/quotecart/CreateQuoteCartUseCase.kt

```diff
@@ -11,19 +11,19 @@ import com.hedvig.app.util.LocaleManager
 import com.hedvig.app.util.apollo.safeQuery
 
 class CreateQuoteCartUseCase(
-    val apolloClient: ApolloClient,
-    localeManager: LocaleManager,
-    marketManager: MarketManager,
+    private val apolloClient: ApolloClient,
+    private val localeManager: LocaleManager,
+    private val marketManager: MarketManager,
 ) {
 
-    private val mutation = CreateOnboardingQuoteCartMutation(
+    private fun mutation() = CreateOnboardingQuoteCartMutation(
         localeManager.defaultLocale().toString(),
         marketManager.market?.toGraphQLMarket() ?: Market.SWEDEN
     )
 
     suspend operator fun invoke(): Either<ErrorMessage, QuoteCartId> {
         return apolloClient
-            .mutate(mutation)
+            .mutate(mutation())
```

Comment:
I'm not entirely sure whether Koin does eager or lazy instantiation of all of its dependencies, but the mutation would indeed be created once the class was instantiated.

The bytecode is indeed equivalent - what's your motivation for preferring the `val`-variant?

Reply:
I'd guess it does it eagerly, since a `lazy` alternative to `single` exists, but it doesn't matter I guess, since we also want it to be refreshed with the latest values coming from the two managers. Even if it'd be lazy those could change later so regardless your change was important.

My motivation was just that it made the mutation *feel* like a local field which just clicked better in my head. But with that said, in this case retrieving it does indeed do two calls to the two managers which definitely feels more of what a function would do. So I'm probably wrong in this case the more I think about it. My point with this basically was that it's a nice catch 😅